wordpress-mobile/WordPress-iOS
# [Jetpack] Enable Domains purchases for wpcom sites in adhoc builds

This PR enables the Domains dashboard and purchasing flow for the Jetpack app, only for wpcom sites and only for adhoc builds.

Requires associated WordPressKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/497

|   |   |   |   |
|---|---|---|---|
| ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-22 at 13 54 56](https://user-images.githubusercontent.com/4780/164718576-d234524b-a81c-4801-adf4-b8f90187306a.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-22 at 13 54 00](https://user-images.githubusercontent.com/4780/164718583-74af7577-dc50-4a03-9cc2-9a4dd3d62d9b.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-22 at 13 54 08](https://user-images.githubusercontent.com/4780/164718650-bea4d191-d244-41a5-a35e-ba9b48a17e2f.png) | ![Simulator Screen Shot - iPhone 13 Pro - 2022-04-22 at 13 54 45](https://user-images.githubusercontent.com/4780/164718665-b4120f7c-ad9a-44d3-91bd-f7311a0c217d.png) |

**To test**

- Build and run the Jetpack target
- View the Menu section of My Site for a number of sites – try to verify WordPress.com sites, self-hosted non-Jetpack and self-hosted Jetpack.
- For WordPress.com sites, you should see the Domains item in the My Site menu, provided that you have site management permissions.
- Navigate into Domains when it's present and ensure you can see the site's current domain listed.
- Attempt the entire domains purchase flow. It's recommended to use the sandbox store – for full information about setting this up, read #p2-pbArwn-2GN. You should be able to successfully complete the flow, and the Domains Dashboard should update and reflect the new domain.
- Also verify that the Domains section isn't visible in the WordPress app.

## Regression Notes

1. Potential unintended areas of impact

Feature being accessible in unintended builds of the app or for non wpcom sites.

2. What I did to test those areas of impact (or what existing automated tests I relied on)4. 

I manually tested the WordPress app and viewed other site types (e.g. self hosted, Jetpack), and verified the feature flag settings.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
